WebDriverAgentMac/WebDriverAgentLib/Commands/FBSessionCommands.m

```diff
@@ -87,7 +88,13 @@ + (NSArray *)routes
   if (nil == bundleID) {
     session = [FBSession initWithApplication:nil];
   } else {
-    XCUIApplication *app = [[XCUIApplication alloc] initWithBundleIdentifier:bundleID];
+    NSString *appPath = requirements[AM_APP_PATH_CAPABILITY];
+    XCUIApplication *app;
+    if (appPath == nil) {
+      app = [[XCUIApplication alloc] initWithBundleIdentifier:bundleID];
+    } else {
+      app = [[XCUIApplication alloc] initPrivateWithPath:appPath bundleID:bundleID];
```

Comment:
should we validate if appPath is actually a valid path to an app bundle and throw an exception if it is not?

Reply:
I think we can skip validation here since it happens inside [XCUIApplication  initPrivateWithPath:bundleID].
If a path is not valid it will throw an error with message: "Failed to determine bundle identifier for '/Users/alex/program/out/SomeApp.app'. The application may be damaged or incomplete."